dace/libraries/torch/dispatchers/cpp_torch_extension.py

```diff
@@ -0,0 +1,740 @@
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
+"""Code generation for PyTorch C++ dispatched operators."""
+import copy
+import dataclasses
+from distutils import sysconfig
+import itertools
+import logging
+import operator
+import os
+import sys
+from typing import List, Tuple, Callable, Optional, Dict, Union
+
+import dace.library
+import numpy as np
+import torch
+import hashlib, operator
+from torch.utils.cpp_extension import load as torch_load
+import dace
+from dace import dtypes as dt, data
+from dace.codegen import targets, compiler
+from dace.codegen.codeobject import CodeObject
+from dace.codegen.compiled_sdfg import CompiledSDFG
+from dace.codegen.prettycode import CodeIOStream
+from dace.codegen.common import sym2cpp
+
+from dace.autodiff import BackwardResult
+from dace.libraries.torch.environments import PyTorch
+from dace.util import is_cuda, platform_library_name
+
+from dace.libraries.torch.dispatchers.common import DaCeMLTorchFunction, compile_and_init_sdfgs, get_arglist
+
+log = logging.getLogger(__name__)
+
+_REPLACED_CTYPES = {dace.int64: "int64_t", dace.uint64: "uint64_t", dace.float16: "at::Half"}
+
+
+def torch_ctype(dtype: dace.typeclass) -> str:
+    """Convert a DaCe type to the corresponding PyTorch C++ type string.
+
+    Args:
+        dtype: The DaCe typeclass to convert
+
+    Returns:
+        The corresponding C++ type string for PyTorch
+    """
+    if isinstance(dtype, dace.pointer):
+        # assuming pointers are 64 bit
+        ctype = "int64_t"
+    elif dtype in _REPLACED_CTYPES:
+        ctype = _REPLACED_CTYPES[dtype]
+    else:
+        ctype = dtype.ctype
+    return ctype
+
+
+_TYPECLASS_TO_TORCH_DTYPE_STR = {
+    dt.bool: "kBool",
+    dt.int8: "kInt8",
+    dt.uint8: "kUInt8",
+    dt.int16: "kInt16",
+    dt.int32: "kInt32",
+    dt.int64: "kInt64",
+    dt.float16: "kFloat16",
```

Comment:
bfloat16 is sorely missing from a lot of the dictionaries here. We should probably add a type to dtypes.py (please make a note for a separate PR for bfloat16, the float8 variants, [and mxfp*?])